src/core/document.js

```diff
@@ -688,6 +731,20 @@ class Page {
     return shadow(this, "_parsedAnnotations", parsedAnnotations);
   }
 
+  get _idToRefMap() {
+    const promise = this._parsedAnnotations.then(annotations => {
+      const idToRefMap = new Map();
+      for (const annotation of annotations) {
+        if (!annotation) {
```

Comment:
Please make sure that the `ref`-property actually exists, note a previous comment above.
```suggestion
        if (!annotation?.ref) {
```